### PR TITLE
Fix lists after last merge

### PR DESCRIPTION
**Pull request by siemiatj**
Bunch of different things stopped working after last refactoring. We need to be more diligent next time, as it should be easily noticeable. Also none of my comment from the refactor PR was ever addressed.

Fixed:
- closing the dropdown on mouse selection
- opening the dropdown with arrow down when the field is focused
- weird props error
- improved focused state indication

### CODE REVIEW

**Comment by PabloSichert** [comment]
Which comments are you referring to? I can't remember anything being still open and it wasn't intentional otherwise

File: src/components/widget/SelectionDropdown.js [hunk 1]
```diff
@@ -129,9 +130,11 @@ export default class SelectionDropdown extends Component {
 
     switch (event.key) {
       case 'ArrowUp':
+        event.preventDefault();
         navigate(true);
         break;
       case 'ArrowDown':
+        event.preventDefault();
```
Comment (PabloSichert) [inline src/components/widget/SelectionDropdown.js]: `preventDefault` is already called [here](https://github.com/metasfresh/metasfresh-webui-frontend/blob/master/src/components/widget/SelectionDropdown.js#L153)

**Comment by siemiatj** [comment]
Yeah, I guess it's an overlook on all sides. I left some comments on the List/Lookup PR as far as I remember. And didn't do a great job reviewing this. Removing onOutsideClick handler was an obvious thing and shouldn't have missed that. This gave us a minor headache on Friday.

File: src/components/widget/SelectionDropdown.js [hunk 2]
```diff
@@ -14,19 +15,7 @@ export default class SelectionDropdown extends Component {
         size: PropTypes.number.isRequired,
       }),
     ]).isRequired,
-    selected: (props, propName) => {
```
Comment (siemiatj) [inline src/components/widget/SelectionDropdown.js]: @PabloSichert can you throw some light on this part ? It looks like a huge overcomplication for something that should be pretty straightforward. As far as I know we expect `selected` to be an object, but I might be missing something. This on the other hand looks like it wants to handle different cases and work for different props (thus the propName I suppose) but will only work for `selected` prop and from my testing it always gets an object or `null`.

Comment (PabloSichert) [reply src/components/widget/SelectionDropdown.js]: It checks if `selected` is either one of `options` or `null/undefined`. So where you got the errors, something different was passed in, which should prevent the component from working properly.

No, it's only supposed to work for "selected" - you are right in that it should rather have been hardcoded there

**Comment by PabloSichert** [comment]
> Removing onOutsideClick handler was an obvious thing and shouldn't have missed that.

I removed it since it wasn't needed anymore after refactoring. Could you give me a link on w101 where functionality is broken? I might have missed it

**Comment by siemiatj** [comment]
@PabloSichert it was not possible to select anything from the dropdown or close them with mouse.

**Comment by PabloSichert** [comment]
> it was not possible to select anything from the dropdown or close them with mouse.

I can't reproduce that. Could you give me a link where mouse interaction is broken?

**Comment by siemiatj** [comment]
https://w101.metasfresh.com:8443/window/540345

```
Pick -> Pick to new HU -> Click on Packing item -> click outside => dropdown keeps opening
```

**Pull request merged**